src/controllers/auth.controller.js

```diff
@@ -0,0 +1,192 @@
+import bcrypt from 'bcrypt';
+import { User } from '../models/user.js';
+import { profileService } from '../services/profile.service.js';
+import { jwtService } from '../services/jwt.service.js';
+import { ApiError } from '../exeptions/api.error.js';
+import { tokenService } from '../services/token.service.js';
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPattern = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!emailPattern.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+
+  await profileService.register(name, email, hashedPassword);
+
+  res.send({
+    message: 'Registration successful. Please check your email for activation.',
+  });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  user.activationToken = null;
+  user.save();
```

Comment:
Consider using 'await user.save()' to ensure the save operation completes before proceeding. This will prevent potential issues with asynchronous operations.